### PR TITLE
 Ignore missing sourcemap if referenced file does not exist

### PR DESCRIPTION
**Pull request by Turbo87**
Resolves https://github.com/ember-cli/broccoli-uglify-sourcemap/issues/53

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -98,7 +98,12 @@ UglifyWriter.prototype.processFile = function(inFile, outFile, relativePath, out
 
     if (srcURL.existsIn(src)) {
       let url = srcURL.getFrom(src);
-      sourceMap.content = JSON.parse(fs.readFileSync(path.join(path.dirname(inFile), url)));
+      let sourceMapPath = path.join(path.dirname(inFile), url);
+      if (fs.existsSync(sourceMapPath)) {
+        sourceMap.content = JSON.parse(fs.readFileSync(sourceMapPath));
+      } else {
+        console.warn(`[WARN] (broccoli-uglify-sourcemap) "${url}" referenced in "${relativePath}" could not be found`);
```
Comment (rwjblue) [inline index.js]: Can you modify this to use `this.console.warn` (then initialize `this.console = options.console || console` in the constructor)? There is at least one other usage of `console.warn` also.

The idea is to have ember-cli-uglify pass in a “custom console” which uses our `UI` object and avoiding needing to manually check for `--silent`...

Comment (Turbo87) [reply index.js]: yeah, I noticed that too, but wanted to address it in a separate PR. your point about honoring `--silent` is valid though, this should probably be something like `} else if (!this.silent) {`.

Comment (Turbo87) [reply index.js]: @rwjblue adjusted

File: test/__snapshots__/test.js.snap [hunk 2]
```diff
@@ -45,6 +46,9 @@ function third(){
   "something.css": "#id {
     background: white;
 }",
+  "with-broken-sourcemap.js": "function meaningOfLife(){throw new Error(42)}
+//# sourceMappingURL=with-broken-sourcemap.map",
+  "with-broken-sourcemap.map": "{\\"version\\":3,\\"sources\\":[\\"0\\"],\\"names\\":[\\"meaningOfLife\\",\\"Error\\"],\\"mappings\\":\\"AAAA,SAASA,gBAEP,MAAM,IAAIC,MADa\\",\\"file\\":\\"with-broken-sourcemap.js\\"}",
```
Comment (xg-wang) [inline test/__snapshots__/test.js.snap]: Is there a way we can output `"sources: ["with-broken-sourcemap.js]` instead of `sources: ["0"]`?

Comment (Turbo87) [reply test/__snapshots__/test.js.snap]: I don't know. can you explain why that would be useful or more correct than what we currently have?

**Pull request merged**